### PR TITLE
Bootstraps a backend for the app

### PR DESCRIPTION
**Pull request by jessex**
## Description of the change

Sorry for the large PR, but fortunately most of the changes in here come from moving fixture data around!

This stands up a backend for the app, copying heavily from the backend of pulse-dashboard. This is a simple Node/Express server hosted in App Engine that fetches metric files from a GCS bucket in response to requests from the frontend. Some important notes on this initial implementation:
* There is no authentication against the API - despite previous statements I have made, I couldn't think of an auth approach that seemed reasonable and worthwhile here, given that the app is publicly available anyway. We can discuss this further soon if there are concerns
* There _is_ authentication from the server to GCS - `.gitignored` service account credential files live in the repo and are deployed up to App Engine, with the appropriate credential file chosen from env vars
* There is a demo mode available - taken directly from the implementation in pulse-dashboard, you can skip going all the way to GCS and instead get files directly from the repository, which is helpful for a number of use cases. We know this form of demo mode comes with some flaws but it's good enough pre-launch
* It uses the same general data flow as pulse-dashboard - I have an in-review design doc that addresses pulse-dashboard's need for a redesign in order to deal with arbitrarily complex metric data, but the data needs for public-dashboard are thankfully quite simple, so we don't bother trying to skate where the data puck is going, not yet
* The server is directly in this repository - same as pulse-dashboard, again. One of the points of my aforementioned design doc is that we should eventually consolidate and extract a standalone Metric API Service for use by these two apps and future apps (both internal and external), but for now this is the quickest path
* It is hosted in the same GCP projects as the server for pulse-dashboard - instead of creating brand new staging and production GCP projects for public-dashboard, I set this up as _services_ inside of the staging and production prijects for pulse-dashboard

This PR includes:
* The `/server` directory with nearly all backend code
* `server.js` in the root of the repo where the actual entrypoint to the backend resides
* Configuration for the environment variables and deploys (example files, that is -- the real stuff is in the shared vault for distribution among the team)
* Consolidated frontend code for making calls to the API
* Slight updates to the existing pages to use this calling code instead of importing test data
* Replacing the frontend-bundled `test_data` with the backend-bundled `demo_data`
  * Also, importantly, changing the fixture files from JSON to JSON Lines, which is what the live buckets actually include
* Code of Conduct and Contribution Guideline docs, which I noticed were missing
* Dependency updates
* Updated scripts in `package.json` which change how you run the app locally
* An updated README

I've deployed the backend to staging just to prove that I can do so and then hit its API, but have not done a production backend deploy or any frontend deploys. You can run this end-to-end against staging data locally via `yarn dev`, though!

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/utils/metricsClient.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+async function callMetricsApi(endpoint) {
+  try {
+    const response = await fetch(
+      `${process.env.REACT_APP_API_URL}/api/${endpoint}`
+    );
+
+    const responseData = await response.json();
+    return responseData;
+  } catch (error) {
+    console.log(error);
```
Comment (macfarlandian) [inline src/utils/metricsClient.js:32]: should this be a call to the `logger` util instead?

Comment (jessex) [reply src/utils/metricsClient.js:32]: Ah, yes. Yes it should.

File: server.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+app.use(cors());
+
+const port = process.env.PORT || 3001;
+app.set("port", port);
+
+app.use(morgan("dev"));
+app.use(helmet());
+
+app.get("/api/:stateCode/parole", api.parole);
```
[the remark this answers is not in the document]
Comment (jessex) [reply server.js:36]: Changed to tenantId across the server.

File: README.md [hunk 3]
```diff
@@ -1,100 +1,139 @@
...
 
-See the section about [deployment](https://facebook.github.io/create-react-app/docs/deployment) for more information.
+When running locally, you can run the app in demo mode to point the app to static data contained in `server/core/demo_data`. This is useful for debugging issues that materialize under specific data circumstances, for demonstrating the tool without exposing real data, for development when you don't have Internet access, and other use cases.
 
-### Deploys
+You can launch in demo mode locally via: `./run_in_demo_mode.sh`
 
-The frontend of the app is deployed to Firebase. To have deploy access, you need to be an admin on the Firebase/GCP account.
+Running via that command is important because environment variables are required for both the frontend and backend servers. Running with only one or the other in demo mode produces a fairly broken experience.
+
```
[the remark this answers is not in the document]
Comment (jessex) [reply README.md:68]: Good idea!

**Comment by jessex** [comment]
Rebased against master, responded to all feedback, CI is passing. Going to merge this in! You'll need to make a couple of changes on your in-flight PR, @macfarlandian, but let me know if you need any assistance!

**Pull request merged**